static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -42,9 +49,12 @@ export class DownloadTemplateComponent implements OnInit {
                     component.ngZone.run(() => { // Need to do this using NgZone since we're calling a third party API
                         component.templateDirectoryInfoRef = component.db.object('template-directory/' + params.id);
                         component.templateRenderInfoRef = component.db.object('template-render-info/' + params.id);
+                        component.templateRatingsInfoRef = component.db.object('template-ratings/' + params.id);
+                        component.templateRatingsInfoDatabaseRef = firebase.database().ref('template-ratings/' + params.id);
```

Comment:
Why are we storing a reference to the entirety of the template's rating info? We're potentially fetching an amount of information that grows linearly with the number of users.

Reply:
We need this in order to get a snapshot of the database. That is how queries are performed, how else can we query without getting a reference? Also, it needs to be a firebase.database().ref because that is what the documentation for transactions uses.